### ISSUE TITLE
correctness: simplify_ring cost function flaw allows area to grow

### ISSUE THREAD

**Issue by anchapin** [anchapin/matchline#359]
## Summary
The `simplify_ring` algorithm in `geometry_simplify.py` has a correctness bug in its greedy cost function. The algorithm removes vertices in order of smallest `_tri_area` (triangle area of vertex with neighbors), but this metric does not predict whether the simplified polygon will have smaller or larger area than the original.

## The Bug
```python
# Line ~365 in geometry_simplify.py
# Cost of removing vertex i:
area_loss = _tri_area(prev[i], i, nxt[i])
```

`area_loss` is the *local* area change — the area of the triangle formed by vertex `i` and its two neighbors. But removing a vertex with small `area_loss` can:
1. Change the overall path in a way that increases the convex hull area
2. Introduce concavities that alter the winding order enough to change area
3. Create a polygon whose `shapely.Polygon.area` differs significantly from the cumulative sum of `_tri_area` removals

## Why This Violates Correctness
The function contract (per `docs/geometry_simplify.md`) states the simplified ring should have area "close to" the original. A simplified ring with *more* area than the original is not a simplification — it's a distortion.

## Impact Beyond Tests
`validate.py` uses `simplify_ring` for conservation law checks. If `simplify_ring` can increase area, then:
- `FootprintAreaConservation` check could produce wrong results
- The review queue could accept incorrect extractions

## Required Fix
The algorithm must guarantee `simplified_area <= original_area * (1 + tol)` for the given tolerance. This may require:
1. Re-computing actual polygon area after each removal (not using `_tri_area` as proxy)
2. Backtracking when area grows
3. A different algorithm entirely (e.g., Visvalingam-Whyatt or similar that uses actual area)

## Labels
correctness
sig: geometry-simplify

**Comment by anchapin** [comment]
Duplicate of #329
